### ISSUE TITLE
Custom criteria syntax is completely asinine

### ISSUE THREAD

**Issue by WarpRules** [Xjph/EliteObservatory#46]
I tried to create a very simple custom criteria: A previously unmapped earth-like body. Sounds simple, right? I literally spent an hour trying to come up with the correct syntax to express that, and had to give up because I couldn't figure it out. And I'm a long-time programmer by profession. Both the completely asinine obscure syntax as well the completely lacking documentation made it near impossible. The program just kept giving me a completely obscure and unhelpful syntax error message.

(Also, my intent was to do the same also for water worlds, ammonia worlds and terraformables. It looked to me like the same dozen-or-so lines of insane XML code would need to be repeated for every one of them because I don't think the syntax supports any kind of "or" list.)

There has to be a better and simpler way than this.

Not only is using XML asinine in itself, but the way in which it's being used here is ten times more asinine. Even with XML a much simpler, easier and much more expressive syntax is possible. (Of course optimally the entire XML would be ditched and something simpler and more straightforward would be used.)

**Comment by Xjph** [comment]
Yes, it is. Which is why [the current version of Observatory](https://github.com/Xjph/ObservatoryCore) uses Lua instead of this XML abomination.
